File: src/main/java/io/github/jho951/assignment/job/service/RequestHashService.java
package io.github.jho951.assignment.job.service;

import java.nio.charset.StandardCharsets;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;

import org.springframework.stereotype.Service;

@Service
public class RequestHashService {

    public String hashImageUrl(String imageUrl) {
        try {
            MessageDigest messageDigest = MessageDigest.getInstance("SHA-256");
            byte[] digest = messageDigest.digest(("imageUrl=" + imageUrl).getBytes(StandardCharsets.UTF_8));
            StringBuilder builder = new StringBuilder(digest.length * 2);
            for (byte value : digest) {
                builder.append(String.format("%02x", value));
            }
            return builder.toString();
        }
        catch (NoSuchAlgorithmException exception) {
            throw new IllegalStateException("SHA-256 algorithm is not available", exception);
        }
    }
}

File: src/main/kotlin/io/github/jho951/assignment/job/service/RequestHashService.kt
package io.github.jho951.assignment.job.service

import java.nio.charset.StandardCharsets
import java.security.MessageDigest
import java.security.NoSuchAlgorithmException
import org.springframework.stereotype.Service

@Service
class RequestHashService {

    fun hashImageUrl(imageUrl: String): String {
        try {
            val messageDigest = MessageDigest.getInstance("SHA-256")
            val digest = messageDigest.digest("imageUrl=$imageUrl".toByteArray(StandardCharsets.UTF_8))
            val builder = StringBuilder(digest.size * 2)
            for (value in digest) {
                builder.append(String.format("%02x", value))
            }
            return builder.toString()
        } catch (exception: NoSuchAlgorithmException) {
            throw IllegalStateException("SHA-256 algorithm is not available", exception)
        }
    }
}
